packages/backend/src/apps/delay/actions/delay-until/index.ts

```diff
@@ -39,21 +53,42 @@ const action: IRawAction = {
       ? new String(delayUntilTime).trim()
       : defaultTime
 
-    const delayTimestamp = generateTimestamp(
+    let delayTimestamp = generateTimestamp(
       delayUntilString,
       delayUntilTimeString,
     )
 
+    let dataItem = {
+      delayUntil: delayUntilString,
+      delayUntilTime: delayUntilTimeString,
+    }
+
+    /**
+     * RETRY: we check and only allow manual retries for failures due to:
+     * - invalid timestamp
+     * - delay until timestamp entered is in the past
+     */
+    const isRetry = await isValidRetry($)
+
     if (isNaN(delayTimestamp)) {
-      throw new StepError(
-        'Invalid timestamp entered',
-        'Check that the date or time entered is of a valid format.',
-        $.step.position,
-        $.app.name,
-      )
+      if (isRetry) {
```

Comment:
i think for now, let's only allow retry if the timestamp is valid and is in the past. For invalid timestamps, let's let the user fix it. And subsequently force retry now if it's in the past